### PR TITLE
chore: remove Unit::AddAura() func overloads and simplify signature

### PR DESCRIPTION
**Pull request by natrist**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Simplify the Unit API and reduce redundancy in code

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- The anti-cheat module will need to be updated to reflect the API changes

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pklloveyou** [comment]
I don't recommend doing this.
After PR
    `A adds a SPELL for B` Become `B adds a SPELL for B`.

File: src/server/scripts/Commands/cs_gm.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ class gm_commandscript : public CommandScript
         }
         else
         {
-            _player->AddAura(VISUAL_AURA, _player);
```
Comment (Kitzunu) [inline src/server/scripts/Commands/cs_gm.cpp:213]: But in places like this, when we add the aura to the same "caster" I think this change is good

File: src/server/game/Spells/SpellEffects.cpp [hunk 2]
```diff
@@ -6100,11 +6100,11 @@ void Spell::SummonGuardian(uint32 i, uint32 entry, SummonPropertiesEntry const*
         {
             // Inferno, Warlock summon spell
             case 1122:
-                caster->AddAura(61191, summon);
```
[the remark this answers is not in the document]
Comment (heyitsbench) [reply src/server/game/Spells/SpellEffects.cpp:6103]: This is also my concern, for the record.

**Comment by sudlud** [comment]
https://github.com/TrinityCore/TrinityCore/commit/8e9d2cdf01929f513e37eccbfdea952aa04e78f6

Just leaving this here as reference where this was originally implemented.

File: src/server/game/Entities/Unit/Unit.cpp [hunk 3]
```diff
@@ -19113,50 +19113,43 @@ void Unit::GetPartyMembers(std::list<Unit*>& TagUnitMap)
...
-    if (Aura* aura = Aura::TryRefreshStackOrCreate(spellInfo, effMask, target, this))
-    {
-        aura->ApplyForTargets();
-        return aura;
+        if (IsImmunedToSpellEffect(spellInfo, i))
+            effectMask &= ~(1 << i);
     }
-    return nullptr;
+
+    Aura* auraObject = Aura::TryRefreshStackOrCreate(spellInfo, effectMask, this, this);
```
Comment (dedmen) [inline src/server/game/Entities/Unit/Unit.cpp:19140]: This looks incorrect.

Ref https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Spells/Auras/SpellAuras.cpp#L326

Previously "caster" was the object casting the aura. Now its the same as the one receiving it.


The caster is relevant here: 
https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Entities/Unit/Unit.cpp#L4305
And here:
https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Spells/Auras/SpellAuras.cpp#L409



If the aura is not singlestack
https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Entities/Unit/Unit.cpp#L4324
Then it does matter who the caster is, that information is lost in this refactor.

I haven't  checked what the caster info stored in Aura class is used for. But this alrady looks wrong.

Comment (natrist) [reply src/server/game/Entities/Unit/Unit.cpp:19140]: Thanks for the feedback.
I should have tagged my PR as a draft. I am well aware of the issues but the bigger problem is none of this code is elegant, performant or concise and the 'owner' in this context is completely irrelevant.

**Comment by Nyeriah** [comment]
Marking as abandoned since this PR hasn't received any updates in a long while and is still a draft. If there's interest in continuing this PR please leave a message and we will reopen it

**Pull request closed without merging**